del_section/src/comp.php

```diff
@@ -0,0 +1,278 @@
+	
+		));
+		$output4 = curl_exec($ch4);
+
+  		$info4 = curl_getinfo($ch4, CURLINFO_HTTP_CODE);
+		$output_json2 = json_decode($output4);
+		curl_close($ch4);
+
+		if ($info4 !== 200){
+			echo "<p>ERROR: There was an error getting the draft section information:</p><pre>" . var_export($output, true) . "</pre>";
+			continue;
+		} else {
+			echo "    Got section draft information</br></br>";
+		}
+
+		$assoc_listid = $output_json2->included[0]->id;
+		echo "    list_id: " . $assoc_listid . "</br>";
+		$title = $output_json2->included[0]->attributes->title;
+		echo "    Title: " . $title . "</br>";
+		$etag = $output_json2->included[0]->meta->list_etag;
+		echo "    ETag: " . $etag . "</br>";
+		
+		fwrite($myfile, $title . "\t");
+		fwrite($myfile, $assoc_listid . "\t");
+		fwrite($myfile, $title . "\t");
+		fwrite($myfile, $barc . "\t");
+
+	if ($shouldWritetoLive == "true") {
+
+	//**************DELETE_SECTION***************
+	$patch_url = 'https://rl.talis.com/3/' . $shortCode . '/draft_sections/' . $barc;
+
+	$input = '	{
+					"meta": {
+						"list_etag": "' . $etag . '",
+						"list_id": "' . $assoc_listid . '"
+					}
+				}';
+
+	//**************POST_THE_DELETE************
+
+	$ch2 = curl_init();
+
+	curl_setopt($ch2, CURLOPT_URL, $patch_url);
+	curl_setopt($ch2, CURLOPT_CUSTOMREQUEST, 'DELETE');
+	curl_setopt($ch2, CURLOPT_RETURNTRANSFER, true);
+	curl_setopt($ch2, CURLOPT_HTTPHEADER, array(
+		
+		"X-Effective-User: $TalisGUID",
+		"Authorization: Bearer $token",
+		'Cache-Control: no-cache'
+	));
+
+	curl_setopt($ch2, CURLOPT_POSTFIELDS, $input);
+
+	
+	$output2 = curl_exec($ch2);
+	$info2 = curl_getinfo($ch2, CURLINFO_HTTP_CODE);
+
+	curl_close($ch2);
+	if ($info2 !== 200){
+		echo "<p>ERROR: There was an error deleting the section:</p><pre>" . var_export($output2, true) . "</pre>";
+		fwrite($myfile, "Section not deleted - failed" . "\t");
+		continue;
+	} else {
+		echo "    Deleted section $barc from list $assoc_listid</br>";
+		fwrite($myfile, "Section deleted successfully" . "\t");
+	}
+
+	//************GRAB**AN**ETAG**AGAIN*************
+
+	$list_lookup = 'https://rl.talis.com/3/' . $shortCode . '/draft_lists/' . $assoc_listid;
+
+	$ch5 = curl_init();
+
+	curl_setopt($ch5, CURLOPT_URL, $list_lookup);
+	curl_setopt($ch5, CURLOPT_RETURNTRANSFER, 1);
+	curl_setopt($ch5, CURLOPT_HTTPHEADER, array(
+
+		"X-Effective-User: $TalisGUID",
+		"Authorization: Bearer $token",
+		'Cache-Control: no-cache'
+
+	));
+	$output5 = curl_exec($ch5);
+	$info5 = curl_getinfo($ch5, CURLINFO_HTTP_CODE);
+	$output_json3 = json_decode($output5);
+	curl_close($ch5);
+
+	$etag2 = $output_json3->data->meta->list_etag;
+	echo "    Updated ETag: " . $etag2 . "</br>";
+	echo "    ---------------------------------------------------";
+	echo "</br>";
+
+	if ($shouldPublishLists === TRUE) {
+		//**************PUBLISH**LIST***************
+		$patch_url2 = 'https://rl.talis.com/3/' . $shortCode . '/draft_lists/' . $assoc_listid . '/publish_actions';
```

Comment:
This needs to be using the bulk publish lists method.
This endpoint is the public publish queue.

Incidentally. using the bulk publish queue means you don't need to get an etag again and so will mean you don't need to do the extra API request to get the list just above this.

Reply:
endpoint & request body updated.